scripts.js

```diff
@@ -28,10 +28,37 @@ document.getElementById('file-input').onchange = function (e) {
   );
 };
 
+document.getElementById('get-url').onclick = function (e) {
+  e.preventDefault();
+  const fileUrl = document.getElementById('file-url').value;
+  document.getElementById('file-url-error').style.display = "none";
+
+  // Check if url is valid
+  if (!domainValid(fileUrl)) {
+    document.getElementById('file-url-error').style.display = "block";
+  } else {
+    fetch(fileUrl)
+    .then(response => response.blob())
```

Comment:
looks good, just tested going to a reasource that doesn't exist and it wont produce an error message in that case.
are you able to add a .catch to handle timeouts or/and handle not found response and add the error to that?

regards Alistair